documentation/src/main/asciidoc/user_guide/execute_grid.adoc

```diff
@@ -0,0 +1,163 @@
+cache manager. There are no guarantees as to if the execution is attempted to be interrupted or not. However
+when the timeout occurs any `Consumer` or `Future` will be completed passing back a `TimeoutException`.
+This value can be overridden by ivoking the
+link:{javadocroot}/org/infinispan/manager/ClusterExecutor.html#timeout-long-java.util.concurrent.TimeUnit-[timeout]
+method and supplying the desired duration.
+
+==== Single Node Submission
+
+Cluster Executor can also run in single node submission mode instead of submitting the command
+to all nodes it will instead pick one of the nodes that would have normally received the command
+and instead submit it it to only one. Each submission will possibly use a different node to
+execute the task on. This can be very useful to use the ClusterExecutor as a
+`java.util.concurrent.Executor` which you may have noticed that ClusterExecutor implements.
+
+[source,java]
+.SingleNode.java
+----
+   EmbeddedCacheManager manager = ...;
+   manager.executor().singleNodeSubmission().submit(...)
+----
+
+===== Predicate Based Filtering
+
+We also allow the target node to be chosen by the user by any means using a `Predicate` that
+will filter out which nodes can be considered for execution. Note this can also be combined
+with Topology filtering at the same time to allow even more fine control of where you code
+is executed within the cluster.
+
+[source,java]
+.Predicate.java
+----
+   EmbeddedCacheManager manager = ...;
+   // Just filter
+   manager.executor().filterTargets(a -> a.equals(..)).submit(...)
+   // Filter only those in the desired topology
+   manager.executor().filterTargets(ClusterExecutionPolicy.SAME_SITE, a -> a.equals(..)).submit(...)
+----
+
+===== Failover
+
+When running in single node submission it may be desirable to also allow the Cluster Executor
+handle cases where an exception occurred during the processing of a given command by retrying
+the command again.
+When this occurs the Cluster Executor will choose a single node again to resubmit the command to
+up to the desired number of failover attempts. Note the chosen node could be any node that passes
+the topology or predicate check. Failover is enabled by invoking the overridden
+link:{javadocroot}/org/infinispan/manager/ClusterExecutor.html#singleNodeSubmission-int-[singleNodeSubmission]
+method. The given command will be resubmitted again to a single node until either
+the command completes without exception or the total submission amount is equal to the provided
+failover count. If however a `org.infinispan.util.concurrent.TimeoutException` is encountered
+either from the command or by the command timing out as defined in `timeout` then it will
+terminate the failover process and complete any pending CompletableFutures with the
+exception.
+
+
+==== Example: Dynamically Start and Stop Clustered Cache
+This example shows how you can use the ClusterExecutor to dynamically start and stop a cache.
```

Comment:
Will this example be still relevant once we have cluster wide cache starts? I'm wondering if we should have a different example whose lifetime might be longer...